tidy3d/components/medium.py

```diff
@@ -1267,6 +1336,257 @@ def elements(self) -> Dict[str, IsotropicMediumType]:
+    >>> cond = [[0.1, 0, 0], [0, 0, 0], [0, 0, 0]]
+    >>> anisotropic_dielectric = FullyAnisotropicMedium(
+    >>>    permittivity=perm, conductivity=cond
+    >>>)
+    """
+
+    permittivity: RealTensor = pd.Field(
+        [[1, 0, 0], [0, 1, 0], [0, 0, 1]],
+        title="Permittivity",
+        description="Relative permittivity tensor.",
+        units=PERMITTIVITY,
+    )
+
+    conductivity: RealTensor = pd.Field(
+        [[0, 0, 0], [0, 0, 0], [0, 0, 0]],
+        title="Conductivity",
+        description="Electric conductivity tensor. Defined such that the imaginary part "
+        "of the complex permittivity at angular frequency omega is given by conductivity/omega.",
+        units=CONDUCTIVITY,
+    )
+
+    @pd.validator("permittivity", always=True)
+    def permittivity_spd_and_ge_one(cls, val):
+        """Check that provided permittivity tensor is symmetric positive definite
+        with eigenvalues >= 1.
+        """
+
+        if not np.allclose(val, np.transpose(val), atol=1.0e-10):
+            raise ValidationError("Provided permittivity tensor is not symmetric.")
+
+        if np.any(np.linalg.eigvals(val) < 1 - 1.0e-10):
+            raise ValidationError("Main diagonal of provided permittivity tensor is not >= 1.")
+
+        return val
+
+    @pd.validator("conductivity", always=True)
+    def conductivity_ge_zero_and_commutes(cls, val, values):
+        """Check that the symmetric part of conductivity tensor commutes with permittivity tensor
+        (that is, simultaneously diagonalizable) with eigenvalues >= 0.
+        """
+
+        perm = values.get("permittivity")
+        cond_sym = 0.5 * (val + np.transpose(val))
+
+        if not np.allclose(
+            np.abs(np.matmul(perm, cond_sym) - np.matmul(cond_sym, perm)), 0, atol=1.0e-10
+        ):
+            raise ValidationError(
+                "Main directions of conductivity and permittivity tensor do not coincide."
+            )
+
+        if np.any(np.linalg.eigvals(cond_sym) < -fp_eps):
+            raise ValidationError("Main diagonal of provided conductivity tensor is not >= 0.")
+
+        return val
+
+    @classmethod
+    def from_diagonal(cls, xx: Medium, yy: Medium, zz: Medium, rotation: RotationType):
+        """Construct a fully anisotropic medium by rotating a diagonally ansisotropic medium.
+
+        Parameters
+        ----------
+        xx : Medium
+            Medium describing the xx-component of the diagonal permittivity tensor.
+        yy : Medium
+            Medium describing the yy-component of the diagonal permittivity tensor.
+        zz : Medium
+            Medium describing the zz-component of the diagonal permittivity tensor.
+        rotation : RotationType
+                Rotation applied to diagonal permittivity tensor.
+
+        Returns
+        -------
+        :class:`FullyAnisotropicMedium`
+            Resulting fully anisotropic medium.
+        """
+
+        permittivity_diag = [
```

Comment:
if we're paying by the line, this could be another option
```py
permittivity_diag = np.diag((xx.permittivity, yy.permittivity, zz.permittivity)).tolist()
```
same for conductivity. Yours is probably clear enough though.

Reply:
changed to a variation of what you suggested